trieste/bayesian_optimizer.py

```diff
@@ -312,29 +343,40 @@ def optimize(
                     f" {OBJECTIVE!r}, got keys {datasets.keys()}"
                 )
 
-            acquisition_rule = cast(AcquisitionRule[S, SP], EfficientGlobalOptimization())
+            acquisition_rule = cast(AcquisitionRule[SP], EfficientGlobalOptimization())
 
         models = map_values(create_model, model_specs)
         history: list[Record[S]] = []
 
+        if trust_region is not None:
+            trust_region_applied = trust_region(self._search_space)
+
+            if trust_region_state is None:
+                trust_region_state = trust_region_applied.default_state
+
         for step in range(num_steps):
             if track_state:
-                history.append(Record(datasets, models, acquisition_state))
+                history.append(Record(datasets, models, trust_region_state))
 
             try:
                 if track_state:
                     models = copy.deepcopy(models)
-                    acquisition_state = copy.deepcopy(acquisition_state)
+                    trust_region_state = copy.deepcopy(trust_region_state)
 
-                if step == 0 and fit_intial_model:
+                if step == 0 and fit_initial_model:
                     for tag, model in models.items():
                         dataset = datasets[tag]
                         model.update(dataset)
                         model.optimize(dataset)
 
-                query_points, acquisition_state = acquisition_rule.acquire(
-                    self._search_space, datasets, models, acquisition_state
-                )
+                if trust_region is None:
+                    acquisition_space = self._search_space
+                else:
+                    trust_region_state, acquisition_space = trust_region_applied.acquire(
+                        datasets, models
+                    )(cast(S, trust_region_state))
```

Comment:
Is the cast just to avoid the None? If not, why not move the defaulting down here? Probably clearer that way anyway.

```suggestion
                    )(trust_region_state if trust_region_state is not None else trust_region_applied.default_state)
```

Reply:
will do

Reply:
fyi changing it like this means the stored state will be `None` for the first run (if the arg is `None`). Still works fine, just slightly differently